web-marketplace/src/pages/Projects/Projects.normalizers.ts

```diff
@@ -1,40 +1,58 @@
 import { AllCreditClassQuery } from 'generated/sanity-graphql';
+import { SKIPPED_CLASS_ID } from 'lib/env';
 
 import { CreditClassWithMedata } from './hooks/useFetchCreditClasses';
+import { UNREGISTERED_PATH, UNREGISTERED_PROJECTS } from './Projects.constants';
+import { CreditClassFilter, ProjectWithOrderData } from './Projects.types';
 
 type NormalizeCreditClassesFilterParams = {
+  allProjects: ProjectWithOrderData[];
   sanityCreditClassesData?: AllCreditClassQuery;
   creditClassesWithMetadata?: CreditClassWithMedata[];
+  haveOffChainProjects: boolean;
 };
 
 type NormalizeCreditClassFiltersResponse = {
   creditClassFilters: CreditClassFilter[];
 };
 
-export type CreditClassFilter = {
-  name: string;
-  path: string;
-  isCommunity: boolean;
-};
-
 export const normalizeCreditClassFilters = ({
+  allProjects,
   creditClassesWithMetadata,
   sanityCreditClassesData,
+  haveOffChainProjects,
 }: NormalizeCreditClassesFilterParams): NormalizeCreditClassFiltersResponse => {
   const sanityCreditClassIds = sanityCreditClassesData?.allCreditClass.map(
     sanityCreditClass => sanityCreditClass.path,
   );
 
-  const creditClassFilters =
-    creditClassesWithMetadata?.map(({ creditClass, metadata }) => {
-      const isCommunity = !sanityCreditClassIds?.includes(creditClass.id);
-
-      return {
-        name: metadata?.['schema:name'] ?? creditClass.id,
-        path: creditClass.id,
-        isCommunity,
-      };
-    }) ?? [];
+  const creditClassesIdsWithProjects = [
+    ...new Set(allProjects.map(project => project.creditClassId)),
+  ];
+
+  const creditClassFilters: CreditClassFilter[] =
+    creditClassesWithMetadata
+      ?.filter(
+        ({ creditClass }) =>
+          creditClass.id !== SKIPPED_CLASS_ID &&
+          creditClassesIdsWithProjects.includes(creditClass.id),
```

Comment:
filtering credit classes without any projects so they don't show up in the filters side bar